lib/src/presentation/widgets/custom_text_field.dart

```diff
@@ -253,7 +254,12 @@ class CustomTextFieldState<S extends CustomTextField> extends State<S> {
   bool get showCustomButton => widget.useCustomButton && hasText;
 
   void onControllerTextChanged() {
-    onChanged(controller.text);
+    if (controller.text.isNotEmpty) {
+      startedTyping = true;
+    }
+    if (startedTyping) {
+      onChanged(controller.text);
+    }
```

Comment:
Here I think the issue is more about the state being refreshed instead of the `onChanged()` being called... the same behavior can be if `startedTyping` is used in the condition for `refreshState`